CMake/cdat_modules_extra/git_clone.sh.in

```diff
@@ -1,4 +1,10 @@
 #!/bin/sh
 
-cd @CMAKE_INSTALL_PREFIX@
-"@GIT_EXECUTABLE@" clone --depth 1 -b @BRANCH@ @GIT_URL@ @GIT_TARGET@
+cd "@CMAKE_INSTALL_PREFIX@"
+"@GIT_EXECUTABLE@" clone --no-checkout --depth 1 -b @BRANCH@ @GIT_URL@ "@GIT_TARGET@"
+cd "@GIT_TARGET@"
+if [ "$("@GIT_EXECUTABLE@" cat-file -t @BRANCH@)" = tag ]; then
+    "@GIT_EXECUTABLE@" checkout @BRANCH@
+else
+    "@GIT_EXECUTABLE@" checkout origin/@BRANCH@
```

Comment:
Note that the local branch `refs/heads/@BRANCH@` will still be created with upstream information (but not checked out); this is because of `clone -b`.